### PR TITLE
Add support for ElastiCache nodes

### PR DESCRIPTION
**Pull request by reedloden**
Support ElastiCache nodes (both Redis and Memcached)

### CODE REVIEW

**Comment by reedloden** [comment]
@itmecho This one is rebased and ready for review.

File: internal/awsdata/elasticache.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+				Row: inventory.Row{
+					UniqueAssetIdentifier:     fmt.Sprintf("%s-%s", aws.StringValue(c.CacheClusterId), aws.StringValue(n.CacheNodeId)),
+					Virtual:                   true,
+					Public:                    false,
+					DNSNameOrURL:              aws.StringValue(n.Endpoint.Address),
+					BaselineConfigurationName: aws.StringValue(c.CacheParameterGroup.CacheParameterGroupName),
+					Location:                  region,
+					AssetType:                 AssetTypeElastiCacheNode,
+					HardwareMakeModel:         aws.StringValue(c.CacheNodeType),
+					// TODO: SoftwareDatabaseVendor
```
Comment (itmecho) [inline internal/awsdata/elasticache.go]: Do we know if c.Engine could be used as this value?

**Review by itmecho — COMMENTED** [review]
Thanks! Just one quick comment

**Comment by itmecho** [comment]
Oh also, I merged #94 so this will need rebasing again

File: internal/awsdata/elasticache.go (hunk 1 above)
Comment (reedloden) [reply internal/awsdata/elasticache.go]: Yeah, close enough. Redis was the one I was a bit concerned about, since it's technically "Redis Labs", but 🤷 .

**Comment by reedloden** [comment]
> Oh also, I merged #94 so this will need rebasing again

This has been done, so good to merge now.

File: internal/awsdata/elasticache.go (hunk 1 above)
Comment (itmecho) [reply internal/awsdata/elasticache.go]: We could do something like this above and then use that variable?
```go
vendor := aws.StringValue(c.Engine)
if vendor == "redis" {
    vendor = "Redis Labs"
}
```

Comment (reedloden) [reply internal/awsdata/elasticache.go]: https://en.wikipedia.org/wiki/Redis#History makes it seem like it is "sponsored" by Redis Labs, versus owned. The more correct thing is probably just saying "Salvatore Sanfilippo" (the actual author), but I think just leaving it as "redis" is fine, imho, especially as being an open source project. Thoughts?

Comment (itmecho) [reply internal/awsdata/elasticache.go]: Works for me!

**Review by itmecho — APPROVED** [review]

**Pull request merged**